src/mode.jl

```diff
@@ -0,0 +1,59 @@
+"""
+    AbstractMode
+
+Abstract supertype for the traits identifying differentiation modes.
+
+# Subtypes
+
+- [`FiniteDifferencesMode`](@ref)
+- [`ForwardMode`](@ref)
+- [`ReverseMode`](@ref)
+- [`ForwardOrReverseMode`](@ref)
+- [`SymbolicMode`](@ref)
+"""
+abstract type AbstractMode end
+
+"""
+    mode(ad::AbstractADType)
+
+Return the differentiation mode of `ad`, as a subtype of [`AbstractMode`](@ref).
+"""
+function mode end
+
+"""
+    FiniteDifferencesMode
+
+Trait for AD choices that rely on [finite differences](https://en.wikipedia.org/wiki/Finite_difference).
+"""
+struct FiniteDifferencesMode <: AbstractMode end
```

Comment:
Why is this not just forward mode? Finite Differences is just jvps so it's structurally the same

Reply:
Because ADTypes used to have four modes. But I agree that finite differences and forward are not different (they are handled the same by DI). Still I think it makes sense to split, and downstream users can reunite them if they want, as I did

Reply:
I don't understand what it's supposed to mean then. How would it even be defined? If you make this a part of forward, then it's jvp vs vjp vs element-wise, which has a clear definition. But this is a piece that is jvp, so should it be a subtype of forward mode?

Reply:
Again, I stuck to the original ADTypes design, but if you think that the distinction makes no sense (and I agree) then I'll remove it

Reply:
I'd remove it.

Reply:
okay